### PR TITLE
Python scripts: Use PDM and virtualenv to manage dependencies

### PR DESCRIPTION
**Pull request by shoogle**
Python 3.12 and later [prevents](https://stackoverflow.com/questions/75608323) users from installing packages system-wide with Pip, as this can create "dependency hell" when different packages depend on different versions of other packages.

The recommended solution is to run each Python program/library inside its own virtual environment that includes only the specific packages needed for that program/library.

[PDM](https://pdm-project.org/) and [virtualenv](https://virtualenv.pypa.io/
) are tools to manage virtual environments and install packages within them. (See [comparison to similar tools](https://dev.to/frostming/a-review-pipenv-vs-poetry-vs-pdm-39b4).)

---

### Set up

1. Install [Pipx](https://pipx.pypa.io/) (e.g. `brew install pipx`).
2. Use Pipx to install PDM (`pipx install pdm`).
3. Run `pdm install` in MuseScore's root directory.

The final command creates a virtual environment (`.venv` folder) and installs the dependencies mentioned in MuseScore's `pyproject.toml` file (added by this PR).

### Running scripts

Simply run the Python scripts as you normally would. E.g.

```Bash
share/instruments/update_instruments_xml.py
```

Boilerplate code is added to the top of scripts that need to use modules from the virtual environment. (This is only necessary for scripts that use modules not present in Python's Standard Library, such as the [Requests module](https://docs.python-requests.org/en/latest/index.html)).

### Adding new dependencies

Basically, use `pdm add <package>` instead of `pip install <package>`.

This installs the package to your own virtual environment and also modifies the `pyproject.toml` and `pdm.lock` files in the repository. You should commit the changes to those files.

Other developers can pick up the new dependency by running `pdm install` again on their machine.

If writing a new script, you should also add the boilerplate code from `update_instruments_xml.py` to ensure the virtual environment gets loaded.

---

### Installing packages with Pip (not recommended)

If you don't want to use virtual environments, you can ignore this PR and use:

```
pip install --user <package> --break-system-packages
```

The boilerplate code will load these modules if no virtual environment is detected.

### CODE REVIEW

**Review by igorkorsukov — APPROVED** [review]

File: share/instruments/update_instruments_xml.py [hunk 1]
```diff
@@ -4,6 +4,25 @@
 def eprint(*args, **kwargs):
     print(*args, **kwargs, file=sys.stderr)
 
+if __name__ == '__main__' and sys.prefix == sys.base_prefix:
```
Comment (cbjeukendrup) [inline share/instruments/update_instruments_xml.py:7]: I assume we do this only in this file because this is the only file that uses a non-pre-installed package (requests), is that right?

**Review by cbjeukendrup — APPROVED** [review]

File: share/instruments/update_instruments_xml.py (hunk 1 above)
Comment (shoogle) [reply share/instruments/update_instruments_xml.py:7]: There are a few other Python files using modules not in the standard library (requests and [markdown](https://pypi.org/project/Markdown/)).

```
src/framework/draw/thirdparty/freetype/freetype-2.13.1/tests/scripts/download-test-fonts.py
buildscripts/ci/release/correct_release_info.py
buildscripts/ci/learn/make_youtube_playlist_info.py
```

Now that the method has been approved in principle, I'll add it to the buildscripts too.

I won't add it to the one in freetype since that's thirdparty.

**Review by cbjeukendrup — APPROVED** [review]

**Comment by shoogle** [comment]
Added documentation at https://github.com/musescore/MuseScore/wiki/Running-Python-scripts
